sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/DocumentResult.java

```diff
@@ -64,4 +67,16 @@ public TextAnalyticsError getError() {
     public boolean isError() {
         return isError;
     }
+
+    /**
+     * Throw a {@link TextAnalyticsException} if result has isError true and when a non-error property was accessed.
+     */
+    void throwExceptionIfError() {
```

Comment:
Should this accept a `ClientLogger`? Won't everything calling into this use the `DocumentResult.class` logger?